Mastodon/Supporting Files/SceneDelegate.swift

```diff
@@ -219,4 +210,93 @@ extension SceneDelegate {
 
         return true
     }
+    
+    func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
+        // Determine who sent the URL.
+        if let urlContext = URLContexts.first {
+            handleUrl(context: urlContext)
+        }
+    }
+    
+    private func showComposeViewController() {
+        if coordinator?.tabBarController.topMost is ComposeViewController {
+            logger.debug("\((#file as NSString).lastPathComponent, privacy: .public)[\(#line, privacy: .public)], \(#function, privacy: .public): composing…")
+        } else {
+            if let authContext = coordinator?.authContext {
+                let composeViewModel = ComposeViewModel(
+                    context: AppContext.shared,
+                    authContext: authContext,
+                    destination: .topLevel
+                )
+                _ = coordinator?.present(scene: .compose(viewModel: composeViewModel), from: nil, transition: .modal(animated: true, completion: nil))
+                logger.debug("\((#file as NSString).lastPathComponent, privacy: .public)[\(#line, privacy: .public)], \(#function, privacy: .public): present compose scene")
+            } else {
+                logger.debug("\((#file as NSString).lastPathComponent, privacy: .public)[\(#line, privacy: .public)], \(#function, privacy: .public): not authenticated")
+            }
+        }
+    }
+    
+    private func handleUrl(context: UIOpenURLContext) {
+        let sendingAppID = context.options.sourceApplication
+        let url = context.url
+
+        if !UIApplication.shared.canOpenURL(url) { return }
+
+        print("source application = \(sendingAppID ?? "Unknown")")
+        print("url = \(url)")
+        
+        switch url.host {
+        case "post":
+            showComposeViewController()
+        case "profile":
+            let components = url.pathComponents
+            if components.count == 2 && components[0] == "/" {
+                let addr = components[1]
+                let tokens = addr.components(separatedBy: "@")
+                if tokens.count != 2 { return }
+                let username = tokens[0]
+                let host = tokens[1]
+                if let authContext = coordinator?.authContext {
```

Comment:
When using the suggested initializer in `RemoteProfileViewModel` you could just do this here:

```suggestion
                if let authContext = coordinator?.authContext {
                    let profileViewModel = RemoteProfileViewModel(
                        context: AppContext.shared,
                        authContext: authContext,
                        acct: components[1]
                    )
                    self.coordinator?.present(
                        scene: .profile(viewModel: profileViewModel),
                        from: nil,
                        transition: .show
                    )
                }
 ```